File: src/2432. The Employee That Worked on the Longest Task.py
class Solution:
    def hardestWorker(self, n: int, logs: List[List[int]]) -> int:
        longest_task = logs[0][1]
        id = logs[0][0]
        for i in range(1, len(logs)):
            if logs[i][1] - logs[i - 1][1] > longest_task:
                longest_task = logs[i][1] - logs[i - 1][1]
                id = logs[i][0]
            elif logs[i][1] - logs[i - 1][1] == longest_task:
                id = min(id, logs[i][0])
        return id

File: src/2432. The Employee That Worked on the Longest Task.cpp
class Solution {
public:
    int hardestWorker(int n, vector<vector<int>>& logs) {
        int longestTask = logs[0][1],
            id = logs[0][0];
        for (int i = 1; i < logs.size(); ++i) {
            if (logs[i][1] - logs[i - 1][1] > longestTask) {
                longestTask = logs[i][1] - logs[i - 1][1];
                id = logs[i][0];
            } else if (logs[i][1] - logs[i - 1][1] == longestTask) {
                id = min(id, logs[i][0]);
            }
        }
        return id;
    }
};